specs/derivation.md

```diff
@@ -649,32 +636,83 @@ The system configuration is updated with L1 log events whenever the L1 epoch ref
 In the *Engine Queue* stage, the previously derived `PayloadAttributes` structures are buffered and sent to the
 [execution engine][g-exec-engine] to be executed and converted into a proper L2 block.
 
-The engine queue maintains references to two L2 blocks:
+The stage maintains references to three L2 blocks:
 
+- The [finalized L2 head][g-finalized]: everything up to and including this block can be fully derived from the
```

Comment:
nit: Link is not rendering. Maybe g-finalized is off? I'm not exactly sure.

Reply:
nice catch, fixed.